dts/bindings/serial/infineon,cat1-uart.yaml

```diff
@@ -0,0 +1,67 @@
+# Copyright (c) 2018, Cypress
+# Copyright (c) 2020, ATL Electronics
+# Copyright (c) 2022 Cypress Semiconductor Corporation (an Infineon company) or
+# an affiliate of Cypress Semiconductor Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+description: Infineon CAT1 UART
+
+compatible: "infineon,cat1-uart"
+
+include: [uart-controller.yaml, pinctrl-device.yaml]
+
+properties:
+  reg:
+    required: true
+
+  interrupts:
+    required: true
+
+  peripheral-id:
+    type: int
+    description: peripheral ID
+    required: true
```

Comment:
where/why is this used?

Reply:
we use it to get uart/scb block instance:
```
#define INFINEON_CAT1_UART_INIT(n)							     \
...
		.hw_resource = {							     \
				.type = CYHAL_RSC_SCB,					     \
				.block_num = DT_INST_PROP(n, peripheral_id),		     \
			},	
...
```
this uses for Uart initialization `cyhal_uart_init_cfg()`  in `ifx_cat1_uart_init`  function.